packages/engine/src/output/buffer/util.rs

```diff
@@ -1,42 +1,74 @@
-use std::path::PathBuf;
-
 use memory::shared_memory::MemoryId;
 
-use crate::{
-    output::{buffer::RELATIVE_PARTS_FOLDER, error::Result},
-    proto::ExperimentId,
-};
+use crate::{output::buffer::RELATIVE_PARTS_FOLDER, proto::ExperimentId};
+
+// TODO: move this to top-level
+#[derive(PartialEq, Eq, Clone)]
+pub enum EngineExitStatus {
+    Success,
+    Error,
+}
 
 // TODO: move this to top-level
 /// Shared memory cleanup in the process hard crash case.
 /// Not required for pod instances.
-pub fn cleanup_experiment(experiment_id: &ExperimentId) -> Result<()> {
-    MemoryId::clean_up(experiment_id)?;
-
-    // TODO: We don't want to be deleting the parts files by default. We should figure out what to
-    //   do with this.
-    // remove_experiment_parts(experiment_id).or_else(|e| {
-    //     let err_string = e.to_string();
-    //     if err_string.contains("kind: NotFound") {
-    //         Ok(())
-    //     } else {
-    //         Err(Error::from(format!(
-    //             "Error while trying to remove experiment parts folder: {err_string}"
-    //         )))
-    //     }
-    // })?;
-
-    Ok(())
+pub fn cleanup_experiment(experiment_id: &ExperimentId, exit_status: EngineExitStatus) {
+    MemoryId::clean_up(experiment_id);
+
+    // Cleanup python socket files in case the engine didn't
+    let frompy_files = glob::glob(&format!("{experiment_id}-frompy*"));
```

Comment:
Why is this logic happening in this file?

Reply:
Relevant comments: https://github.com/hashintel/hash/pull/477#discussion_r847316801

Reply:
I still don't understand why this function is in `output/buffer/util.rs` though.

Reply:
I don't know why this function is here either xD
It was here, the is a todo to move it, that's all I know.

Reply:
It would be good if these sorts of questions are asked during development. If we're touching a function with a TODO like that, we should be reading the TODO and asking if it's still relevant. Especially if it's a TODO about moving the function and we're greatly expanding what the function does. And especially if we start adding random unrelated enums in.

Reply:
Moved it in https://github.com/hashintel/hash/pull/494, not sure if it's the right place